Core/src/org/sleuthkit/autopsy/datamodel/FileTypesByExtension.java

```diff
@@ -330,24 +331,39 @@ long calculateChildCount() throws TskCoreException {
         }
     }
 
-    private static String createQuery(FileTypesByExtension.SearchFilterInterface filter) {
-        StringBuilder query = new StringBuilder();
-        query.append("(dir_type = ").append(TskData.TSK_FS_NAME_TYPE_ENUM.REG.getValue()).append(")"); //NON-NLS
-        if (UserPreferences.hideKnownFilesInViewsTree()) {
-            query.append(" AND (known IS NULL OR known != ").append(TskData.FileKnown.KNOWN.getFileKnownValue()).append(")"); //NON-NLS
-        }
-        query.append(" AND (NULL"); //NON-NLS
-        for (String s : filter.getFilter()) {
-            query.append(" OR LOWER(name) LIKE LOWER('%").append(s).append("')"); //NON-NLS
+    private String createQuery(FileTypesByExtension.SearchFilterInterface filter) {
+        if (filter.getFilter().isEmpty()) {
+            return "";
```

Comment:
I think returning an empty string is wrong here.  In the intended use case, this gets passed to SleuthkitCase.countFilesWhere() or findAllFilesWhere() where it is used as the predicate expression in the Where clause.  If this is empty, ie  "select count(*) as count from tsk_files where ", it is a SQL syntax error.  I would rather throw an exception or return something that evaluates to true (in both Postgres and SQLite); i think you recommended "1=1" as such an expression previously.